### ISSUE TITLE
Ability to avoid publishing to `all` stream

### ISSUE THREAD

**Issue by Paskid** [RailsEventStore/rails_event_store#673]
Hi,
I was wondering if it was under the scope to have the capacity to not publish event in the `all` stream when using named streams.
In my current use case, the application we are developing has multiple tenants so the `all` stream is simply filling the database uselessly.

Right now what I have done is I overrid `RailsEventStoreActiveRecord::EventRepository`'s  `append_to_stream` method to pass `false` in the `include_global` parameter but I would rather not override the original gem.
```
Rails.configuration.to_prepare do
  Rails.configuration.event_store = RailsEventStore::Client.new repository: CustomEventStoreRepository.new
end

class CustomEventStoreRepository < RailsEventStoreActiveRecord::EventRepository
  def append_to_stream(events, stream, expected_version)
    add_to_stream(normalize_to_array(events), stream, expected_version, false) do |event|
      build_event_record(event).save!
      event.event_id
    end
  end
end
```

I have read the pull request #514 that was started last year and I think it would solve this.
Do you have an idea as to when this could go live or if there could be a way to implement a simple configuration property to use instead of the hard-coded `true` passed to `include_global`?

Thank you

**Comment by pawelpacana** [comment]
> I have read the pull request #514 that was started last year and I think it would solve this.

Yes, that was the intent of this PR

> Do you have an idea as to when this could go live?

My biggest objection was the effort needed to introduce it on an already running system with accumulated events. I intended to document how one could migrate first. The implmentation is there and probably needs only a rebase to current master.

The way you swapped repository with custom one is possibly the best you can do now to achieve this. The risky part is `CustomEventStoreRepository < RailsEventStoreActiveRecord::EventRepository` as it relies on internals that could change per release but I guess you're aware of that.

**Comment by Paskid** [comment]
Hi @pawelpacana,
You are right, I am aware of the risks of my implementation, that is why I was wondering if, as a temporary alternative, a configuration field of some sort could be added to let me disable the `include_global` until pull request #514 is resolved (unless you think it is only a matter of time in which no changes to the `EventRepository` should happen)

What do you suggest would be the best approach time wise?

**Comment by pawelpacana** [comment]
Solution landed on master and set to be released in 2.0
